feeds/src/state/ducks/aggregator/selectors.ts

```diff
@@ -3,16 +3,54 @@ import { createSelector } from 'reselect'
 import { AppState } from 'state'
 import { OracleNode } from '../../../config'
 
-const upcaseOracles = (
-  state: AppState,
+type OracleAddress = string
+type NodeAddress = string
+
+type StateBranch = {
+  aggregator: {
+    oracleNodes: {
+      [address: string]: {
+        name: string
+        nodeAddress: NodeAddress[]
+        oracleAddress: OracleAddress
+        [prop: string]: any
```

Comment:
This was necessary to allow the entire state to be passed in, because TS uses strict object checks.

Reply:
I feel like this should be using `AppState` in the implementation and the `partialAsFull` test helper. There some example usage in https://github.com/smartcontractkit/chainlink/blob/develop/feeds/src/state/ducks/listing/selectors.test.ts#L54

Reply:
Updated